### PR TITLE
Add builder for TargetUrl for building a url with parameters.

### PR DESCRIPTION
**Pull request by stefan-dimitrov**
Issue: #1 

Added a builder to TargetUrl as suggested by @mlesikov in the issue.

Usage example:
```java
String urlTemplateString = "http://sample.com/a/:valueA/b/:valueB/end";
TargetUrl targetUrl = TargetUrl.fromTemplate(urlTemplateString).set("valueA", "123").set("valueB", "678").build();

// result url string is "http://sample.com/a/123/b/678/end"
```

### CODE REVIEW

**Comment by stefan-dimitrov** [comment]
Implemented throwing of exception when not all parameters of the template have been set.
Added appending of query parameters (standard GET parameters: `/url/path?param=value`).

**Comment by mlesikov** [comment]
LGTM.

File: src/main/java/com/clouway/chita/TargetUrl.java [hunk 1]
```diff
@@ -3,12 +3,83 @@
...
 import com.google.common.base.Strings;
 
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
```
Comment (mlesikov) [inline src/main/java/com/clouway/chita/TargetUrl.java]: add documentation - examples here and over the ChitaHttpClient interface

Comment (stefan-dimitrov) [inline src/main/java/com/clouway/chita/TargetUrl.java]: done.

**Pull request merged**